nhrpd/netlink_arp.c

```diff
@@ -93,20 +97,36 @@ static void netlink_neigh_msg(struct nlmsghdr *msg, struct zbuf *zb)
 	if (!c)
 		return;
 
-	if (msg->nlmsg_type == RTM_GETNEIGH) {
-		debugf(NHRP_DEBUG_KERNEL, "Netlink: who-has %s dev %s",
-		       sockunion2str(&addr, buf, sizeof buf), ifp->name);
+	debugf(NHRP_DEBUG_KERNEL,
+	       "Netlink: %s %s dev %s lladdr %s nud 0x%x "
+	       "cache used %u type %u",
+	       (msg->nlmsg_type == RTM_GETNEIGH)
+		       ? "who-has"
+		       : (msg->nlmsg_type == RTM_NEWNEIGH) ? "new-neigh"
+							   : "del-neigh",
+	       sockunion2str(&addr, buf[0], sizeof(buf[0])), ifp->name,
+	       sockunion2str(&lladdr, buf[1], sizeof(buf[1])), ndm->ndm_state,
+	       c->used, c->cur.type);
 
+	if (msg->nlmsg_type == RTM_GETNEIGH) {
 		if (c->cur.type >= NHRP_CACHE_CACHED) {
 			nhrp_cache_set_used(c, 1);
-			netlink_update_binding(ifp, &addr,
```

Comment:
Could you elaborate? The should be the same value. If there's a difference there's a something broken in the code, or you have some other code injecting disagreeing values in the kernel table.

In fact, the `NDA_LLADDR` can be empty in certain cases when the entry is expired, or someone added incorrect route table entry pointing to the device.

I think this change is also incorrect and should be reverted. But I would be happy to hear when exactly `lladdr != vc->remote.nbma` and fix the underlying issues elsewhere.